src/App.js

```diff
@@ -1,7 +1,72 @@
-import React from "react";
+import React, { useEffect, useState } from 'react';
+import jwt_decode from 'jwt-decode';
+import Parse from 'parse';
 
-const App = () => {
-  return <h1>Hello React!</h1>;
-};
+function App() {
+  const [ user, setUser ] = useState({});
 
-export default App;
+  function handleCallbackResponse(response) {
+    console.log("Encoded User Token: " + response.credential);
+    var userObject = jwt_decode(response.credential);
+    console.log(userObject);
+    setUser(userObject);
+    document.getElementById("signInDiv").hidden = true;
+
+    // Log in to Parse Server using Google token
+    Parse.User.logInWith('google', { authData: { id: userObject.sub, access_token: response.credential } })
+      .then(user => {
+        console.log('User logged in through Google:', user);
+      })
+      .catch(error => {
+        console.error('Error logging in through Google:', error);
+      });
+  }
+
+  function handleSignOut(event) {
+    // Log out from Parse Server
+    Parse.User.logOut()
+      .then(() => {
+        console.log('User logged out from Parse');
+        setUser({});
+        document.getElementById("signInDiv").hidden = false;
+      })
+      .catch(error => {
+        console.error('Error logging out:', error);
+      });
+  }
+
+  useEffect(() => {
+    Parse.initialize("parse"); // APP_ID
+    Parse.serverURL = 'https://api.fieldsandgoals.com/parse'; // SERVER_URL
+
+    google.accounts.id.initialize({
+        client_id: "141115154444-ffidu0bon5vvv7e1r0r0p0tqsj0oisq2.apps.googleusercontent.com",
+        callback: handleCallbackResponse,
+    });
+
+    google.accounts.id.renderButton(
+        document.getElementById("signInDiv"),
+        { theme: "outline", size: "large" }
+    );
+
+    google.accounts.id.prompt();
```

Comment:
Move this block of code in useEffect into a function and call that func inside the useEffect.
It is good practice to make code more modular and readable. That way we know what this block of code is trying to do.